azurerm/internal/services/postgres/postgresql_server_resource.go

```diff
@@ -898,8 +898,8 @@ func flattenSecurityAlertPolicy(props *postgresql.SecurityAlertPolicyProperties,
 
 	block["enabled"] = props.State == postgresql.ServerSecurityAlertPolicyStateEnabled
 
-	block["disabled_alerts"] = utils.FlattenStringSlice(props.DisabledAlerts)
-	block["email_addresses"] = utils.FlattenStringSlice(props.EmailAddresses)
+	block["disabled_alerts"] = utils.FlattenStringSliceRemoveEmptyString(props.DisabledAlerts)
+	block["email_addresses"] = utils.FlattenStringSliceRemoveEmptyString(props.EmailAddresses)
```

Comment:
ultimately it looks like this shouldn't be necessary if we update the Expand to not add the element if it's an empty string: https://github.com/terraform-providers/terraform-provider-azurerm/pull/8846/files#diff-6d8a16299e927bb0011aaa89cbb729f9R6-R10 - can we revert this and fix the Expand method instead?